src/Og.php

```diff
@@ -196,17 +194,17 @@ public static function getEntityGroups(EntityInterface $entity, array $states =
 
     /** @var \Drupal\og\Entity\OgMembership $membership */
     foreach ($memberships as $membership) {
-      static::$entityGroupCache[$identifier][$membership->getGroupEntityType()][$membership->id()] = $membership->getGroup();
+      static::$cache[$identifier][$membership->getGroupEntityType()][$membership->id()] = $membership->getGroup();
```

Comment:
Reading this it seems like this loop will do a single entity load for every group, which is not very efficient. In practice there are usually not a large number of groups so this might be not a big deal after all.